test/test.coffee

```diff
@@ -2,12 +2,40 @@ expect = require('chai').expect
 
 _ = require 'lodash'
 Promise = require 'bluebird'
+mimemessage = require 'mimemessage'
+
 PinejsClientCore = require '../core'
 
-PinejsClientCore = PinejsClientCore(_, Promise)
+PinejsClientCore = PinejsClientCore(_, Promise, mimemessage)
 core = new PinejsClientCore()
 exports.test = (expected, params) ->
 	if _.isError(expected)
 		expect(-> core.compile(params)).to.throw(expected.constructor, expected.message)
 	else
 		expect(core.compile(params)).to.equal expected
+
+exports.testBatch = (expected, params) ->
+	_.forEach core.compileBatch(params), (req, key) ->
+		if !_.isArray(expected[key]) then expect(req).to.eql(expected[key])
```

Comment:
Can you put the body on a new line please - this totally confused me at first glance